zone/bot.cpp

```diff
@@ -2263,23 +2263,11 @@ bool Bot::TryNonCombatMovementChecks(Client* bot_owner, const Mob* follow_mob, g
 		if ((!bot_owner->GetBotPulling() || PULLING_BOT) && (destination_distance > GetFollowDistance())) {
 
 			if (!IsRooted()) {
-
 				if (rest_timer.Enabled()) {
 					rest_timer.Disable();
 				}
 
-				bool running = true;
-
-				if (destination_distance < GetFollowDistance() + BOT_FOLLOW_DISTANCE_WALK) {
-					running = false;
-				}
-
-				if (running) {
-					RunTo(Goal.x, Goal.y, Goal.z);
-				}
-				else {
-					WalkTo(Goal.x, Goal.y, Goal.z);
-				}
+				RunTo(Goal.x, Goal.y, Goal.z);
```

Comment:
Bots shouldn't always run, do you have a video of the new behavior change showing walk/run transitions?

Reply:
I don’t have a video of the transition. Not having them run always means they have an even larger chance to lag behind because they take a second to start moving once owner moves then they have to wait for a certain distance to start running.

Reply:
99.999999999% of the time a player has [ Run ] toggled on. When a player moves they switch from a standing position to run. There is no walk transition between standing still and running. Bots should also follow this same rule of movement.

Yes, while it does look more visually graceful for bots to transition from standing still to running and running to stopping by using walk it just creates more distance between the player and the bots due to the distance they need to walk instead of run.

The movement speed increase should help overall. Having walk transitions just adds distance between the player and the bots.

Reply:
The transition part of the code is clearly broke which is why there’s a problem to begin with. It’d be nice to fix the root of the problem and have the graceful transition rather than force run the entire time. What this does is better than the current scenario.